tests/Test/AbstractIntegrationCaseFactory.php

```diff
@@ -224,7 +224,7 @@ private function determineCode(SplFileInfo $file, ?string $code, string $suffix)
     private function parseJson(?string $encoded, array $template = null): array
     {
         // content is optional if template is provided
-        if (!$encoded && null !== $template) {
+        if ((null === $encoded || '' === $encoded) && null !== $template) {
```

Comment:
I've spent some time on this one, the change is technically correct, but isn't that strange that both `null` and empty string leads to the same condition? I wonder if we could refactor the tests (in separate PR of course) to have this condition simpler.

Reply:
It could be `empty($encoded)` but PHPStan's strict rules don't allow it 😅. Yeah, it could be normalised somehow to make it simpler here, but with how it works currently it's proper solution IMHO.